### PR TITLE
Vmo 3373 constrain block width

### PR DESCRIPTION
**Pull request by safydy-r**
[Few scenarios]
- Block has min-width, eg: Block A & Block B
- Black has max-width if the label is too long, eg: Block C
- Exits have max-width if the text is too long, eg: Block C / Exit 2
- Exits are shrunk to preserve the block max-width as possible, eg: Block D/ Exit 1 & 2
- Exits are on min-width in case there are too many exits (All those exits should be shown, meaning the block max-width is extended), eg: Block E

block width constraints

### CODE REVIEW

File: src/components/interaction-designer/Block.vue [hunk 1]
```diff
@@ -158,6 +176,13 @@
 
       ...mapGetters('builder', ['blocksById']),
 
+      labelContainerDynamicStyle() {
+        console.log(`get labelContainerMaxWidth ${this.labelContainerMaxWidth}`)
```
Comment (safydy-r) [inline src/components/interaction-designer/Block.vue:180]: Will clean this

File: src/components/interaction-designer/Block.vue [hunk 2]
```diff
@@ -432,18 +463,18 @@
         border: 1px dashed transparent;
         transition: border-radius 200ms ease-in-out;
 
-        /*flex: auto;*/
-        min-width: 6em;
-        max-width: 140px;
+        min-width: 25px;
+        flex-basis: 100px;
+        flex-shrink: 1;
```
[the remark this answers is not in the document]
Comment (safydy-r) [reply src/components/interaction-designer/Block.vue]: It's using now bootstrap flex utilities

**Review by bzabos — APPROVED** [review]
Nice, thanks @safydy! Looks good, I left a couple final thoughts just to overshare my observations😛 I think we can merge if you're happy with it.

File: src/components/interaction-designer/Block.vue [hunk 3]
```diff
@@ -160,9 +162,24 @@ export default {
...
+    // Update the label container max width
+    const blockExitElement = document.querySelector(`#block\\/${this.block.uuid} .block-exit`) // one exit
+
+    if (!blockExitElement) {
+      return;
+    }
+
+    // This allows us to shrink .block-exit into the min-width before extending the block label container
+    if (LABEL_CONTAINER_MAX_WIDTH < this.block.exits.length * blockExitElement.clientWidth) { // compare with `n x {inner width}`
+      this.labelContainerMaxWidth = this.block.exits.length * blockExitElement.offsetWidth // but set with `n x {outer width}`
```
[the remark this answers is not in the document]
Comment (safydy-r) [reply src/components/interaction-designer/Block.vue:175]: - I observed the issue with shrinking, and fixed it
- about the lagging, I did not notice that really. Anyway, I took the opportunity to improve the performance a bit but avoiding to compute the `labelContainerMaxWidth` in `updated()`, but inside a `watcher` when the block exits length changes

File: src/components/interaction-designer/Block.vue [hunk 4]
```diff
@@ -176,6 +193,12 @@ export default {
 
     ...mapGetters('builder', ['blocksById', 'isEditable']),
 
+    labelContainerDynamicStyle() {
+      return {
+        'max-width': `${this.labelContainerMaxWidth}px`
```
[the remark this answers is not in the document]
Comment (safydy-r) [reply src/components/interaction-designer/Block.vue:198]: Thanks, moved it

**Pull request merged**